src/ast/visitor.rs

```diff
@@ -38,6 +38,9 @@ pub trait Visitor<'a> {
     fn visit_excepthandler(&mut self, excepthandler: &'a Excepthandler) {
         walk_excepthandler(self, excepthandler);
     }
+    fn visit_format_spec(&mut self, format_spec: &'a Expr) {
+        walk_expr(self, format_spec);
+    }
```

Comment:
This is a special method to separate normal JoinedStr and format-spec JoinedStr.